_posts/addons/2017-10-23-tcp-addon.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Scalingo TCP Addon
+modified_at: 2017-10-23 00:00:00
+category: addons
+tags: tcp load balancer addon custom
+---
+
+## Introduction
+
+By default, Scalingo will only let you run application that communicate via
+HTTP or HTTPS. Sometimes this is not enough and you need to have application
+that can accept raw TCP connections.
+
+This addon let you deploy application that can accept raw TCP connections.
+
+## Setup of the Addon
+
+### Provision the Addon
+
+First, you need to add the TCP addon to your application. This can be done
+through the dashboard on the addons section of your app under the `Network`
+category.
+
+### Setup Configuration
+
+In order to use the Scalingo TCP addon, you need to define a new container
+type. This can be done in your [Procfile]({% post_url
+internals/2014-12-01-procfile %}).
+
+Here is a sample Procfile for a Node.js application:
+
+```
+web: npm start
+tcp: npm run start-tcp
+```
+
+### Contact your Application
+
+When your application starts, we define a `PORT` environment variable in your
+container. Your application must listen on this port in order to be accessible
+from the outside. This is the exact same mechanism and the exact same
+environment variable that the one used to expose a HTTP server in a `web`
+container.
+
+Once up and running your application will be accessible at the URL defined by
+the `TCP_URL` environment variable.
+
+This `TCP_URL` looks like:
+`tcp://tcp-gateway-1.eu-fr-agora-1.scalingo.com:20003`
+
+You can find this variable on our dashboard or by using our CLI:
+
+```bash
+scalingo --app my-app env | grep TCP_URL
+```
+
+This URL is also accessible on the addon dashboard.
```

Comment:
This is kind of redundant with the above sentence `You can find this variable on our dashboard`. You can safely remove this one I think.

Reply:
Nop, the first one is the Scalingo's dashboard and the second one is the Irma's dashboard